File: app/src/main/java/com/example/englishapp/core/network/dictionary/dto/DictionaryResponseDto.java
package com.example.englishapp.core.network.dictionary.dto;

import com.google.gson.annotations.SerializedName;

import java.util.Collections;
import java.util.List;

public class DictionaryResponseDto {
    @SerializedName("word")
    private String word;

    @SerializedName("phonetic")
    private String phonetic;

    @SerializedName("phonetics")
    private List<PhoneticDto> phonetics;

    @SerializedName("meanings")
    private List<MeaningDto> meanings;

    public DictionaryResponseDto() {
        this.word = "";
        this.phonetic = "";
        this.phonetics = Collections.emptyList();
        this.meanings = Collections.emptyList();
    }

    // Getters
    public String getWord() { return word; }
    public String getPhonetic() { return phonetic; }
    public List<PhoneticDto> getPhonetics() { return phonetics; }
    public List<MeaningDto> getMeanings() { return meanings; }

    // Setters
    public void setWord(String word) { this.word = word; }
    public void setPhonetic(String phonetic) { this.phonetic = phonetic; }
    public void setPhonetics(List<PhoneticDto> phonetics) { this.phonetics = phonetics; }
    public void setMeanings(List<MeaningDto> meanings) {

        this.meanings = meanings;

    }
}

File: app/src/main/java/com/example/englishapp/core/network/dictionary/dto/DictionaryResponseDto.kt
package com.example.englishapp.core.network.dictionary

import kotlinx.serialization.Serializable

@Serializable
data class DictionaryResponseDto(
    val word: String,
    val phonetic: String? = null,
    val meanings: List<MeaningDto> = emptyList()
)

@Serializable
data class MeaningDto(
    val partOfSpeech: String,
    val definitions: List<DefinitionDto> = emptyList()
)

@Serializable
data class DefinitionDto(
    val definition: String,
    val example: String? = null
)